# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1449930757 18000
#      Sat Dec 12 09:32:37 2015 -0500
# Node ID f2cd811f0f9ea7914dbd55c047bc13383d9fec7f
# Parent  0bb9b4e99ea097df7d652b72e1d05d7bd742d9aa
maint: Refactor C++ calls to print_usage to resemble m-files (2015 code sprint)

* file-io.cc (Ffopen): Move nargin checking to the top of the function.

diff --git a/libinterp/corefcn/file-io.cc b/libinterp/corefcn/file-io.cc
--- a/libinterp/corefcn/file-io.cc
+++ b/libinterp/corefcn/file-io.cc
@@ -626,16 +626,19 @@ When opening a new file that does not ye
 @end deftypefn")
 {
   octave_value_list retval;
 
   retval(0) = -1.0;
 
   int nargin = args.length ();
 
+  if (nargin < 1 || nargin > 4)
+    print_usage ();
+
   if (nargin == 1)
     {
       if (args(0).is_string ())
         {
           // If there is only one argument and it is a string but it
           // is not the string "all", we assume it is a file to open
           // with MODE = "r".  To open a file called "all", you have
           // to supply more than one argument.
@@ -650,43 +653,38 @@ When opening a new file that does not ye
           retval(2) = tmp(2);
           retval(1) = tmp(1);
           retval(0) = tmp(0);
 
           return retval;
         }
     }
 
-  if (nargin > 0 && nargin < 4)
-    {
-      octave_value mode = (nargin == 2 || nargin == 3)
-                          ? args(1) : octave_value ("r");
+  octave_value mode = (nargin == 2 || nargin == 3)
+                      ? args(1) : octave_value ("r");
 
-      octave_value arch = (nargin == 3)
-                          ? args(2) : octave_value ("native");
-
-      int fid = -1;
-
-      octave_stream os = do_stream_open (args(0), mode, arch, "fopen", fid);
+  octave_value arch = (nargin == 3)
+                      ? args(2) : octave_value ("native");
 
-      if (os)
-        {
-          retval(1) = "";
-          retval(0) = octave_stream_list::insert (os);
-        }
-      else
-        {
-          int error_number = 0;
+  int fid = -1;
+
+  octave_stream os = do_stream_open (args(0), mode, arch, "fopen", fid);
 
-          retval(1) = os.error (false, error_number);
-          retval(0) = -1.0;
-        }
+  if (os)
+    {
+      retval(1) = "";
+      retval(0) = octave_stream_list::insert (os);
     }
   else
-    print_usage ();
+    {
+      int error_number = 0;
+
+      retval(1) = os.error (false, error_number);
+      retval(0) = -1.0;
+    }
 
   return retval;
 }
 
 DEFUN (freport, args, ,
        "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} freport ()\n\
 Print a list of which files have been opened, and whether they are open\n\
